### DISCUSSION TITLE
Electrical measurement: Power Factor value

### DISCUSSION THREAD

**Discussion by dduransseau** [Koenkk/zigbee2mqtt#20593] [General]
Hi,

By discussing issue #19705 I figure out that power factor might be wrongly interpreted by Z2M (front-end).
Following [Zigbee Cluster Library](https://zigbeealliance.org/wp-content/uploads/2021/10/07-5123-08-Zigbee-Cluster-Library.pdf) p348 the power factor is represented as

    Contains the single phase or PhaseA, Power Factor ratio in 1/100ths.

So it's divided by 100 by zigbee-herdsman-converter to have a power factor between 0 and 1, which is the correct behavior. However the front-end display it as a percentage which is wrong but a common usage.
I see 3 solution to display the correct value:

- Avoid to display a percentage in front end and just display the value between 0 and 1 with two decimal
- Multiply the value by 100 and let the percentage as it is
- Avoid to divide by 100 the power factor in zigbee-herdsman-converter and let the value as it is

I'm not really convinced by last solution since it broke a bit the zigbee interpretation but that would work.

**Comment by Koenkk** [comment]
I think the correct fix is to remove the [% unit](https://github.com/Koenkk/zigbee-herdsman-converters/blob/f91372db3030021364d2f0fb4a6e3524e0b9578e/src/lib/exposes.ts#L718)

**Comment by dduransseau** [comment]
Agreed that would better match the zigbee definition. However that might imply a regression for tuya device where power_factor is provided as raw value so between -100 and +100, at least for one I have (_TZE200_bkkmqmyo).

**Reply by Koenkk** [reply]
For those case, we can still use the `withUnit()`

**Reply by dduransseau** [reply]
But how we can identify if the exposed power_factor metric come from tuya device or standard zigbee device ?

**Reply by Koenkk** [reply]
In the exposes of the tuya device, use `e.power_factor().withUnit('%')` and remove `withUnit('%')` from `exposes.ts`

**Reply by dduransseau** [reply]
Ok, so yes by updating tuja file. I'll have a look at this and PR.

**Reply by dduransseau** [reply]
PR for this [#6847](https://github.com/Koenkk/zigbee-herdsman-converters/pull/6847)

**Reply by Patricen1** [reply]
Hello,

Just updated to 1.39.1 and issue looks to be still there.
On the attached screenshot, we should have apparent power @392VA,  power @392*0.84 = 329W and reactive power = 392-329 =63W
 ![image](https://github.com/user-attachments/assets/b4709208-8326-4bbc-9604-407c4163902d)

Maybe I'm wrong as this concern is about BITUO TECHNIK SPM01-U01
